src/lib.rs

```diff
@@ -2286,14 +2287,23 @@ impl Connection {
 
         // Generate coalesced packets.
         while left > 0 {
-            let (ty, written) =
-                match self.send_single(&mut out[done..done + left]) {
-                    Ok(v) => v,
+            // When there is Initial packet already and try to add
+            // Short packet in the same datagram, add PADDING frame.
+            if has_initial {
+                if let Ok(packet::Type::Short) = self.write_pkt_type() {
+                    add_padding = true;
+                }
+            }
+
+            let (ty, written) = match self
+                .send_single(&mut out[done..done + left], add_padding)
```

Comment:
It might be better to simply pass the `has_initial` value to `send_single()` so that we can unify the logic in one place inside it, rather than split in two parts, something like:
```rust
if has_initial && pkt_type == packet::Type::Short && left > 1 ...
```
this also means we wouldn't have to duplicate the `write_pkt_type()` call in `send()`, which seems somewhat "cleaner".

Reply:
Updated.